### ISSUE TITLE
clang-tidy: false positive from `modernize-concat-nested-namespaces` on inline namespaces

### ISSUE THREAD

**Issue by vogelsgesang** [llvm/llvm-project#56022]
clang-tidy 14 incorrectly complains about the following input

```cpp
namespace asmjit::inline _abi_1_9::x86 { // <-- it claims that the "x86" namespace could be concatenated, although it is already concatenated
class Assembler;
class Gpq;
}
```

When applying the proposed rewrite, the code is rewritten to

```cpp
namespace asmjit::inline _abi_1_9namespace x86 {
class Assembler;
class Gpq;
}
```
